From c755c36c8b301ca0ae46bc428820ebc3b95f1344 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Fri, 2 Jun 2023 04:26:13 +0200
Subject: [PATCH 22/56] pinephone-pro: Fix pmugrf io voltage

On poweron this register is 0 (as opposed to what TRM says, that it
should be 0x300). Make it 0x300.
---
 board/pine64/pinephone-pro-rk3399/pinephone-pro-rk3399.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/board/pine64/pinephone-pro-rk3399/pinephone-pro-rk3399.c b/board/pine64/pinephone-pro-rk3399/pinephone-pro-rk3399.c
index b6ccbb9c1c4..3347ad6b70a 100644
--- a/board/pine64/pinephone-pro-rk3399/pinephone-pro-rk3399.c
+++ b/board/pine64/pinephone-pro-rk3399/pinephone-pro-rk3399.c
@@ -17,7 +17,8 @@
 
 #define GRF_IO_VSEL_BT565_GPIO2AB 1
 #define GRF_IO_VSEL_AUDIO_GPIO3D4A 2
-#define PMUGRF_CON0_VSEL_SHIFT 8
+#define PMUGRF_CON0_VOLSEL_SHIFT 8
+#define PMUGRF_CON0_VOL_SHIFT 9
 
 #ifndef CONFIG_SPL_BUILD
 int board_early_init_f(void)
@@ -53,7 +54,7 @@ static void setup_iodomain(void)
 		  GRF_IO_VSEL_BT565_GPIO2AB | GRF_IO_VSEL_AUDIO_GPIO3D4A);
 
 	/* Set GPIO1 1.8v/3.0v source select to PMU1830_VOL */
-	rk_setreg(&pmugrf->soc_con0, 1 << PMUGRF_CON0_VSEL_SHIFT);
+	rk_setreg(&pmugrf->soc_con0, 1 << PMUGRF_CON0_VOLSEL_SHIFT | 1 << PMUGRF_CON0_VOL_SHIFT);
 }
 
 int misc_init_r(void)
-- 
2.39.2

